### PR TITLE
Rewrite Error.p.stack

### LINKED ISSUE #30906: Error.prototype.stack: non-standard but universally supported
### MDN URL

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Error/stack

### What specific section or headline is this issue about?

The red Non-standard block at the very top of the page

### What information was incorrect, unhelpful, or incomplete?

It needs clarity, and leading with the red block is misleading because the feature is fully supported by every browser listed at the bottom of the very same page. Yes, it is not in the spec, at least not in [the document linked](https://tc39.es/ecma262/multipage/fundamental-objects.html#sec-error-objects) in the MDN page for the `Error` object. But it is supported everywhere except the more minor mobile browsers, where [caniuse.com lists support as "unknown"](https://caniuse.com/?search=stack).

### What did you expect to see?

I would prefer to see this Non-standard notice at the bottom of the page, directly above the Browser Compatibility section. I would also prefer to see additional text that explains the wide support but lack of a defined standard.  Browser Compatibility already has a warning about non-standard, in addition to this red block. If support is really so wide, maybe the block shouldn't be red?

I submit this as an issue because I don't know what MDN policy is for something like this.  I can submit a PR to make changes to this, if someone authorized here deems it worthy of change.

### Do you have any supporting links, references, or citations?

_No response_

### Do you have anything more you want to share?

_No response_

### MDN metadata

<details>
<summary>Page report details</summary>

* Folder: `en-us/web/javascript/reference/global_objects/error/stack`
* MDN URL: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Error/stack
* GitHub URL: https://github.com/mdn/content/blob/main/files/en-us/web/javascript/reference/global_objects/error/stack/index.md
* Last commit: https://github.com/mdn/content/commit/5c3c25fd4f2fbd7a5f01727a65c2f70d73f1880a
* Document last modified: 2023-09-12T04:14:14.000Z

</details>

### PR DESCRIPTION
**Pull request by Josh-Cena**
Fix #30906

### CODE REVIEW

File: files/en-us/web/javascript/reference/global_objects/error/stack/index.md [hunk 1]
```diff
@@ -9,6 +9,8 @@ browser-compat: javascript.builtins.Error.stack
 
 {{JSRef}} {{Non-standard_Header}}
 
+> **Note:** The `stack` property is de-facto implemented by most, if not all, JavaScript engines. [The standard committee is looking to standardize this feature too](https://github.com/tc39/proposal-error-stacks). You cannot rely on the precise content of the stack string due to implementation inconsistencies, but you can usually assume it exists and use it for debugging purposes.
```
Comment (chrisdavidmills) [inline files/en-us/web/javascript/reference/global_objects/error/stack/index.md:12]:
```suggestion
> **Note:** The `stack` property is de facto implemented by all major JavaScript engines, and [the JavaScript standards committee is looking to standardize it](https://github.com/tc39/proposal-error-stacks). You cannot rely on the precise content of the stack string due to implementation inconsistencies, but you can generally assume it exists and use it for debugging purposes.
```

File: files/en-us/web/javascript/reference/global_objects/error/stack/index.md [hunk 2]
```diff
@@ -24,100 +26,129 @@ Because the `stack` property is non-standard, implementations differ about where
...
+#### SpiderMonkey
+anonymous@filename.js line 1 > Function:1:8
+@filename.js:1:65
+
+====
+@filename.js line 3 > eval:1:1
+@filename.js:3:13
+
+#### V8
+Error: Function failled
```
Comment (chrisdavidmills) [inline files/en-us/web/javascript/reference/global_objects/error/stack/index.md:106]:
```suggestion
Error: Function failed
```

**Review by chrisdavidmills — CHANGES_REQUESTED** [review]
@Josh-Cena thanks for the great work on this! It mostly makes sense; I have left a few suggestions for you to consider.

**Comment by sidewayss** [comment]
I just came across a case where Chrome does leaves `stack` undefined: the clipboard. Can't create a fiddle/pen/playground because reproducing it requires setting a breakpoint on the first line of this block. That shifts focus away from the page during the copy operation, causing an error.  Once it breaks, you step line-by-line (F10) a couple of times until the error is caught:
```
navigator.clipboard.writeText(txt)
    .then (()  => /* insert code here */; )
    .catch(err => alert(`Copy failed: ${err.stack ?? err}`));
```
Note the end of the last line where I fall back to `err.toString()` when `err.stack` is undefined. I just added that bit because of this behavior. Firefox does define `stack` here (though it's only one line deep). So would this be a Chrome bug, or simply a side-effect of not having a standard in place?

**Comment by teoli2003** [comment]
Note that the Clipboard is a weird beast as, in Webidl, it isn't an interface but a namespace. No surprise something strange is happening with it.

File: files/en-us/web/javascript/reference/global_objects/error/stack/index.md [hunk 3]
```diff
@@ -24,100 +26,129 @@ Because the `stack` property is non-standard, implementations differ about where
...
-Starting with Firefox 30 and later containing the column number:
+In V8, the non-standard `Error.captureStackTrace()`, `Error.stackTraceLimit`, and `Error.prepareStackTrace()` APIs can be used to customize the stack trace. Read the [Stack trace API](https://v8.dev/docs/stack-trace-api) in the V8 docs for more information.
 
-```plain
-trace@file:///C:/example.html:9:17
-b@file:///C:/example.html:16:13
-a@file:///C:/example.html:19:13
-@file:///C:/example.html:21:9
-```
+Stack frames can be things other than explicit function calls, too. For example, event listeners, timeout jobs, and promise handlers all begin their own call chain. Source code within {{jsxref("Global_Objects/eval", "eval()")}} and {{jsxref("Function")}} constructor also appear in the stack:
```
[the remark this answers is not in the document]
Comment (Josh-Cena) [reply files/en-us/web/javascript/reference/global_objects/error/stack/index.md:80]: It's quite hard to describe, or maybe I don't understand this deeply enough. I am writing a page for execution contexts (maybe in another year? 😛) https://github.com/mdn/content/issues/5006 and I'll talk more about stacks then.

File: files/en-us/web/javascript/reference/global_objects/error/stack/index.md [hunk 4]
```diff
@@ -24,100 +26,129 @@ Because the `stack` property is non-standard, implementations differ about where
...
+    at eval (eval at <anonymous> (filename.js:1:13), <anonymous>:1:8)
+    at filename.js:1:65
+====
+Error: eval failed
+    at eval (eval at <anonymous> (filename.js:3:13), <anonymous>:1:1)
+    at filename.js:3:13
 ```
 
-Firefox 13 and earlier would instead produce the following text:
+You can also use the `//# sourceURL` directive to name an eval source. For Firefox, see also the [Debug eval sources](https://firefox-source-docs.mozilla.org/devtools-user/debugger/how_to/debug_eval_sources/index.html) docs and the [Naming `eval` Scripts with the `//# sourceURL` Directive](https://fitzgeraldnick.com/2014/12/05/name-eval-scripts.html) blog post.
```
[the remark this answers is not in the document]
Comment (Josh-Cena) [reply files/en-us/web/javascript/reference/global_objects/error/stack/index.md:115]: I don't know but I think it is Firefox-specific. After all, devtools are always implementation-inconsistent. I just copied this from the paragraph that was in examples.

File: files/en-us/web/javascript/reference/global_objects/error/stack/index.md [hunk 5]
```diff
@@ -24,100 +26,129 @@ Because the `stack` property is non-standard, implementations differ about where
...
-a("first call, firstarg")@file:///C:/example.html:19
-@file:///C:/example.html:21
-```
+## Examples
 
-### Stack of eval'ed code
+### Using the stack property
 
-Starting with Firefox 30, the error stack of code in `Function()` and `eval()` calls, now produces stacks with more detailed information about the line and column numbers inside these calls. Function calls are indicated with `"> Function"` and eval calls with `"> eval"`.
+The following script demonstrates the use of `stack` property. It runs in your browser, so you can check how the stack trace looks in your browser.
```
[the remark this answers is not in the document]
Comment (Josh-Cena) [reply files/en-us/web/javascript/reference/global_objects/error/stack/index.md:121]: The key here is that the stack trace is "from your browser", so if you don't use one of the browsers described above, or if you use a different version, you can verify what its stack structure looks like.

**Comment by Josh-Cena** [comment]
> So would this be a Chrome bug, or simply a side-effect of not having a standard in place?

More likely a bug. But the only way to tell is by sending an issue and getting their reply.

**Comment by sidewayss** [comment]
Chrome bug for clipboard error: https://bugs.chromium.org/p/chromium/issues/detail?id=1510545

File: files/en-us/web/javascript/reference/global_objects/error/stack/index.md (hunk 4 above)
[the remark this answers is not in the document]
Comment (chrisdavidmills) [reply files/en-us/web/javascript/reference/global_objects/error/stack/index.md:115]: Oh, OK. I've edited my suggested change to suit.

File: files/en-us/web/javascript/reference/global_objects/error/stack/index.md (hunk 3 above)
[the remark this answers is not in the document]
Comment (chrisdavidmills) [reply files/en-us/web/javascript/reference/global_objects/error/stack/index.md]: Is this any use: https://stackoverflow.com/questions/10057443/explain-the-concept-of-a-stack-frame-in-a-nutshell#10057535 ?

I wasn't thinking that it needed to be much, but it seems a little odd that there is no definition for terms such as these. I guess when you come to write your dedicated execution contexts page, you can link it to the definitions in there. I'll let this pass for now, as the definition of it is implied enough to not seriously impair understanding.

File: files/en-us/web/javascript/reference/global_objects/error/stack/index.md (hunk 4 above)
[the remark this answers is not in the document]
Comment (chrisdavidmills) [reply files/en-us/web/javascript/reference/global_objects/error/stack/index.md]: Committed.

File: files/en-us/web/javascript/reference/global_objects/error/stack/index.md (hunk 5 above)
[the remark this answers is not in the document]
Comment (chrisdavidmills) [reply files/en-us/web/javascript/reference/global_objects/error/stack/index.md:121]: Edit suggestion updated to suit. I'm not quite so sure about this one, so I'll let you check this before committing.

**Review by chrisdavidmills — APPROVED** [review]
OK, updates look good, so I'm setting this to approved. Nice work @Josh-Cena!

There are just a couple more comments from me for you to check out, and a couple of updated suggestions for you to review. After that we can merge.

**Comment by Josh-Cena** [comment]
LGTM

**Comment by bsmth** [comment]
Thanks, all! Shall we merge this one?

**Comment by chrisdavidmills** [comment]
> Thanks, all! Shall we merge this one?

Yussssss ;-)

**Comment by sidewayss** [comment]
> Chrome bug for clipboard error: https://bugs.chromium.org/p/chromium/issues/detail?id=1510545

fyi - this bug is fixed and heading for release in Chrome 122.  One less gap in the support for Error.prototype.stack.

**Comment by Josh-Cena** [comment]
Great to know!

**Comment by sidewayss** [comment]
Just got Chrome 122 and tested it to confirm that it contains the bug fix.

**Pull request merged**